cpp/src/io/json/read_json.cu

```diff
@@ -238,6 +244,26 @@ table_with_metadata read_batch(host_span<std::unique_ptr<datasource>> sources,
   return device_parse_nested_json(buffer, reader_opts, stream, mr);
 }
 
+constexpr bool can_be_delimiter(char c)
+{
+  // The character list below is from `json_reader_options.set_delimiter`.
```

Comment:
This may be moved into a static function of `json_reader_options` so we won't have to maintain two separate set of non-allowed characters.